posthog/hogql/printer.py

```diff
@@ -448,9 +448,7 @@ def visit_select_query(self, node: ast.SelectQuery):
         limit = node.limit
         if self.context.limit_top_select and is_top_level_query:
             if limit is not None:
-                if isinstance(limit, ast.Constant) and isinstance(limit.value, int):
-                    limit.value = min(limit.value, MAX_SELECT_RETURNED_ROWS)
```

Comment:
this will now just set the limit provided instead of hard clamping everything. hogqlqueryexecutor does the clamping for queries. printer only clamps if there's no limit provided at all

alternatively, for more safety, I could pass in the limit context into printer and handle the limiting in here